api/cat.ts

```diff
@@ -0,0 +1,49 @@
+export async function getCatBreedById(id: string) {
+  const res = await fetch(`https://api.thecatapi.com/v1/breeds/${id}`)
+
+  if (!res.ok) {
+    throw new Error('Failed to fetch data')
+  }
+
+  return res.json()
+}
+
+export async function getBreeds() {
+  const res = await fetch('https://api.thecatapi.com/v1/breeds')
+
+  if (!res.ok) {
+    throw new Error('Failed to fetch data')
+  }
+
+  return res.json()
+}
+
+export async function getCatImageById(id: string) {
+  const res = await fetch(`https://api.thecatapi.com/v1/images/${id}`)
+
+  if (!res.ok) {

```

Comment:
What would happen in the application UI when an error is thrown? What would the user see and how would you handle such kind of errors in a production environment?

Reply:
Next.js has a REALLY great feature for managing loading UI and error handling UI. I've already added the loading feature in this project. So what's next? Just add an `error.tsx` to the nested route and everything's all set. Easy peasy, right?

https://nextjs.org/docs/app/building-your-application/routing/error-handling

Reply:
Magical error handling out of the box always seems good! But what if it was up to you to define it? Do you have any quick ideas on how you'd do it?

Reply:
I'll use `ErrorBoundary` component. Actually Next.js wrap a route segment and its nested children in a `React Error Boundary`